src/main/java/frc/robot/subsystems/ElevatorSubsystem.java

```diff
@@ -0,0 +1,248 @@
+    private DoubleEntry m_elevatorIzSubs;
+    private DoubleEntry m_elevatorFFSubs;
+    private DoublePublisher m_elevatorSetpointPub;
+    private DoublePublisher m_elevatorVelPub;
+    private DoublePublisher m_elevatorFFTestingVolts;
+    private DoubleEntry m_elevatorOffset;
+    private DoublePublisher m_targetAngle;
+    private DoublePublisher m_elevatorPosPub;
+
+    // for elevator ff
+    private DoubleEntry m_elevatorMomentToVoltage;
+
+    //spark absolute encoder
+    private SparkAbsoluteEncoder m_absEncoder;
+    //embedded relative encoder
+    private SparkClosedLoopController m_pidControllerElevator;
+
+    private final TrapezoidProfile.Constraints m_constraints =
+            new TrapezoidProfile.Constraints(Config.ElevatorConfig.MAX_VEL, Config.ElevatorConfig.MAX_ACCEL);
+    private final ProfiledPIDController m_ProfiledPIDController =
+            new ProfiledPIDController(1.6,0.002,40, m_constraints, 0.02);
+
+
+    public static ElevatorSubsystem getInstance() {
+        if (instance == null) {
+            SubsystemChecker.subsystemConstructed(SubsystemType.ElevatorSubsystem);
+            instance = new ElevatorSubsystem();
+        }
+        return instance;
+    }
+
+    private ElevatorSubsystem() {
+        m_elevator = new SparkMax(Config.ElevatorConfig.ELEVATOR_SPARK_CAN_ID, motorType); // creates SparkMax motor controller
+        m_elevator_config = new SparkMaxConfig();
+
+        m_elevator.setCANTimeout(Config.CANTIMEOUT_MS);
+
+        m_elevator_config.smartCurrentLimit(Config.ElevatorConfig.CURRENT_LIMIT);
+        m_elevator_config.inverted(Config.ElevatorConfig.SET_INVERTED);
+        m_elevator_config.idleMode(IdleMode.kBrake);
+        m_elevator_config.voltageCompensation(6);
+        m_elevator_config.softLimit.forwardSoftLimit(Config.ElevatorConfig.elevator_up_limit);
+        m_elevator_config.softLimit.reverseSoftLimit(Config.ElevatorConfig.elevator_down_limit);
+        m_elevator_config.softLimit.forwardSoftLimitEnabled(Config.ElevatorConfig.SOFT_LIMIT_ENABLE);
+        m_elevator_config.softLimit.reverseSoftLimitEnabled(Config.ElevatorConfig.SOFT_LIMIT_ENABLE);
+        m_elevator_config.signals.primaryEncoderPositionPeriodMs(20);
+        m_elevator_config.signals.primaryEncoderVelocityPeriodMs(20);
+        m_elevator_config.encoder.inverted(Config.ElevatorConfig.INVERT_ENCODER);
+        m_elevator_config.encoder.positionConversionFactor(Config.ElevatorConfig.elevatorPositionConversionFactor);
+        m_elevator_config.encoder.velocityConversionFactor(Config.ElevatorConfig.elevatorVelocityConversionFactor);
+        m_elevator_config.absoluteEncoder.zeroOffset(Math.toRadians(Config.ElevatorConfig.elevatorAbsEncoderOffset));
+        m_elevator_config.closedLoop.feedbackSensor(ClosedLoopConfig.FeedbackSensor.kAbsoluteEncoder);
+
+        m_elevator.configure(m_elevator_config, SparkBase.ResetMode.kResetSafeParameters, SparkBase.PersistMode.kPersistParameters);
```

Comment:
You can refer other subsystem, e.g, arm. 
Try to only call ResetMode once at the end, since it is a blocking function.